fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+        call ESMF_ConfigGetAttribute(config=CF, value=lon2,    label ='lon2:',   rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=lat2,    label ='lat2:',   rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=dlon,    label ='dlon:',   rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=dlat,    label ='dlat:',   rc=rc)
+        imo = (lon2-lon1)/dlon + 1
+        jmo = (lat2-lat1)/dlat + 1
+        if (mype == 0) then
+          print *,'lon1=',lon1,' lat1=',lat1
+          print *,'lon2=',lon2,' lat2=',lat2
+          print *,'dlon=',dlon,' dlat=',dlat
+          print *,'imo =',imo, ' jmo=',jmo
+        end if
+      else if (trim(output_grid) == 'lambert_conformal') then
+        call ESMF_ConfigGetAttribute(config=CF, value=cen_lon, label ='cen_lon:',rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=cen_lat, label ='cen_lat:',rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=stdlat1, label ='stdlat1:',rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=stdlat2, label ='stdlat2:',rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=imo,     label ='nx:',     rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=jmo,     label ='ny:',     rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=lon1,    label ='lon1:',   rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=lat1,    label ='lat1:',   rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=dx,      label ='dx:',     rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=dy,      label ='dy:',     rc=rc)
+        if (mype == 0) then
+          print *,'cen_lon=',cen_lon,' cen_lat=',cen_lat
+          print *,'stdlat1=',stdlat1,' stdlat2=',stdlat2
+          print *,'lon1=',lon1,' lat1=',lat1
+          print *,'nx=',imo, ' ny=',jmo
+          print *,'dx=',dx,' dy=',dy
+        endif
+      endif ! output_grid
+
+    endif ! quilting
+!
+    call ESMF_ConfigGetAttribute(config=CF, value=dt_atmos, label ='dt_atmos:',   rc=rc)
+    call ESMF_ConfigGetAttribute(config=CF, value=nfhmax,   label ='nhours_fcst:',rc=rc)
+    if(mype == 0) print *,'af nems config,dt_atmos=',dt_atmos,'nfhmax=',nfhmax
+    call ESMF_TimeIntervalSet(timeStep,s=dt_atmos,rc=rc)
+    call ESMF_ClockSet(clock_fv3,timeStep=timeStep, rc=rc)
+!
+!------------------------------------------------------------------------
+! may need to set currTime for restart
+!
+    call ESMF_ClockGet(clock_fv3, currTIME=CurrTime,  StartTime=startTime,    &
+                       RunDuration=RunDuration, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    StopTime = startTime + RunDuration
+
+! *** read restart time from restart file
+    do i=751,899
+       inquire(i, opened=OPENED)
+       if(.not. OPENED)then
+         io_unit = i
+         exit
+       endif
+    enddo
```

Comment:
This could still fail without an io_unit.  Look into `ESMF_UtilIOUnitGet - Scan for a free I/O unit number`
http://earthsystemmodeling.org/docs/release/ESMF_8_1_1/ESMF_refdoc/node6.html#SECTION0601441400000000000000